src/app/containers/ContactsContainer.js

```diff
@@ -93,38 +93,38 @@ const ContactsContainer = ({ location, history }) => {
         });
     }, [filteredContacts, checkedContacts, contactEmailsMap]);
 
-    if (loadingContactEmails || loadingContacts || loadingUserKeys || loadingContactGroups) {
-        return <Loader />;
-    }
-
-    const getCheckedContactIDs = () => {
+    const checkedContactIDs = useMemo(() => {
         return Object.entries(checkedContacts).reduce((acc, [contactID, isChecked]) => {
             if (!isChecked) {
                 return acc;
             }
             acc.push(contactID);
             return acc;
         }, []);
-    };
+    }, [checkedContacts]);
 
     const getCurrentContactID = () => {
         const [, contactID] = location.pathname.split('/contacts/');
         return contactID;
     };
 
-    const getContactIDsToDelete = () => {
-        const checkedContactIDs = getCheckedContactIDs();
-        if (checkedContactIDs.length) {
+    const activeIDs = useMemo(() => {
+        if (checkedContactIDs && checkedContactIDs.length) {
             return checkedContactIDs;
         }
         const currentContactID = getCurrentContactID();
         if (currentContactID) {
             return [currentContactID];
         }
-    };
+        return [];
+    }, [checkedContactIDs, location.pathname]);
+
+    if (loadingContactEmails || loadingContacts || loadingUserKeys || loadingContactGroups) {
+        return <Loader />;
+    }
 
     const handleDelete = async () => {
-        const contactIDs = getContactIDsToDelete();
+        const contactIDs = activeIDs();
```

Comment:
```suggestion
        const contactIDs = activeIDs;
```

Reply:
yeah, I forgot to retouch that since actually part of that logic is not needed anymore